pkgs/build-support/fetchgit/nix-prefetch-git

```diff
@@ -451,8 +477,13 @@ export GIT_CONFIG_NOSYSTEM=1
 
 if test -n "$builder"; then
     test -n "$out" -a -n "$url" -a -n "$rev" || usage
-    mkdir -p "$out"
-    clone_user_rev "$out" "$url" "$rev"
+    if test -z "$rootDir"; then
+        mkdir -p "$out"
+        clone_user_rev "$out" "$url" "$rev"
+    else
+        clone_user_rev_to_tmpfile "$url" "$rev"
+        mv "$tmpClone/$rootDir" "$out"
+    fi
```

Comment:
Nit: I'd prefer the if/else branches the other way around; I think code is generally easier to read when the positive case is first.